src/plugins/dashboard/public/application/dashboard_app_controller.tsx

```diff
@@ -590,7 +591,8 @@ export class DashboardAppController {
     const refreshDashboardContainer = () => {
       const changes = getChangesFromAppStateForContainerState();
       if (changes && dashboardContainer) {
-        dashboardContainer.updateInput(changes);
+        const sessionId = searchService.session.start();
+        dashboardContainer.updateInput({ ...changes, searchSessionId: sessionId });
```

Comment:
@lizozom, @lukasolson, @ppisljar, I am passing the `sessionId` down the pipeline (https://github.com/elastic/kibana/pull/76889#discussion_r505329158). Is this the correct approach or do we want `sessionService` to be accessed somewhere down the pipe directly? 

--- 

If this is the preferred approach, then what is the need to global `sessionService` ?

Reply:
Ideally, we wouldn't need the global `sessionService` state. We'd just be passing the `sessionId` around. However, I think the intent of the global state was in case we didn't have the necessary pipeline to send the `sessionId` all the way from the application, through embeddables and expression service, to the search service. @lizozom Correct me if I'm wrong.

Reply:
if i remember correctly initially we were discussing accessing the session directly but since we figured we will have searches which we don't want to be part of the session, so we'll need to be passing them down